api/references/when-clause-contexts.md

```diff
@@ -16,37 +16,128 @@ For example, VS Code uses when clauses to enable or disable command keybindings,
+`!foo && bar`                          | `(!foo) && bar`
+<code>!foo \|\| bar </code>            | `(!foo) \|\| bar`
+<code>foo \|\| bar && baz </code>      | <code>foo \|\| (bar && baz)</code>
+<code>!foo && bar \|\| baz </code>     | <code>(!foo && bar) \|\| baz</code>
+<code>!(foo \|\| bar) && baz </code>   | <code>(remains same) !(foo \|\| bar) && baz</code>
+
+### Equality operators
 
-### key-value when clause operator
+You can check for equality of a context key's value against a specified value. Note that the right-hand side is a value and not interpreted as a context key, meaning it is not looked up in the context.
 
-There is a key-value pair match operator for `when` clauses. The expression `key =~ value` treats the right-hand side as a regular expression to match against the left-hand side. For example, to contribute context menu items for all Docker files, one could use:
+Operator   | Symbol | Example
+--------   | ------ | -----
+Equality   | `==`   | `"editorLangId == typescript"` or `"editorLangId == 'typescript'"`
+Inequality | `!=`   | `"resourceExtname != .js"` or `"resourceExtname != '.js'"`
+
+Notes:
+
+- If the value on the right-hand side is a string containing whitespace, it must be wrapped in single-quotes - `"resourceFilename == 'My New File.md'"`.
+- `===` has the same behavior as `==`, and `!==` has the same behavior as `!=`
+
+### Comparison operators
+
+You can compare a context key's value against a number. Note that left- and right-hand side of the operator must be separated by whitespace - `foo < 1`, but not `foo<1`.
+
+Operator | Symbols | Example
+-------- | ------ | -----
+Greater than | `>`, `>=` | `"gitOpenRepositoryCount >= 1"` but not `"gitOpenRepositoryCount>=1"`
+Less than | `<`, `<=` | `"workspaceFolderCount < 2"` but not `"workspaceFolderCount<2"`
+
+### Match operator
+
+(previous name: key-value pair match operator)
+
+Operator | Symbol | Example
+-------- | ------ | -----
+Matches | `=~` | `"resourceScheme =~ /^untitled$\|^file$/"`
+
+There is a match operator (`=~`) for when clauses. The expression `key =~ regularExpressionLiteral` treats the right-hand side as a regular expression literal to match against the left-hand side. For example, to contribute context menu items for all Docker files, one could use:
 
 ```json
    "when": "resourceFilename =~ /docker/"
 ```
 
-## Available contexts
+Notes:
+
+- The right-hand side of the `=~` operator follows the same rules as JavaScript regular expression literals ([reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#creating_a_regular_expression)), except characters need to follow escaping rules both of JSON strings and regular expressions. Hence, using a slash within the regular expression requires double escaping. For example, what is written as `/file:\/\//` in JavaScript needs to become `/file:\\/\\//` when used within a when clause.
```

Comment:
You can leave of the `/en-US` locale in the URL so that non-US non-English readers can hopefully see a localized version of the page.